components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerSnapshotTests.m

```diff
@@ -0,0 +1,213 @@
+@property(nonatomic, strong) MDCTextInputControllerOutlined *textFieldController;
+@end
+
+@implementation MDCTextFieldOutlinedControllerSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.textField = [[MDCTextField alloc] init];
+  self.textFieldController =
+      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];
+}
+
+- (void)tearDown {
+  self.textFieldController = nil;
+  self.textField = nil;
+
+  [super tearDown];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Helpers
+
+- (void)triggerTextFieldLayout {
+  CGSize aSize = [self.textField sizeThatFits:CGSizeMake(300, INFINITY)];
+  self.textField.bounds = CGRectMake(0, 0, aSize.width, aSize.height);
+  [self.textField layoutIfNeeded];
+}
+
+- (void)generateSnapshotAndVerify {
+  [self triggerTextFieldLayout];
+
+  self.testView = [self addBackgroundViewToView:self.textField];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Tests
+
+- (void)testOutlinedTextFieldEmpty {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Single field tests
+
+- (void)testOutlinedTextFieldWithShortPlaceholderText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.placeholderText = MDCTextFieldSnapshotTestsPlaceholderShortTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongPlaceholderText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.placeholderText = MDCTextFieldSnapshotTestsPlaceholderLongTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithShortHelperText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.helperText = MDCTextFieldSnapshotTestsHelperShortTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongHelperText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.helperText = MDCTextFieldSnapshotTestsHelperLongTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithShortErrorText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  [self.textFieldController setErrorText:MDCTextFieldSnapshotTestsErrorShortTextLatin
+                 errorAccessibilityValue:MDCTextFieldSnapshotTestsErrorShortTextLatin];
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongErrorText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  [self.textFieldController setErrorText:MDCTextFieldSnapshotTestsErrorLongTextLatin
+                 errorAccessibilityValue:MDCTextFieldSnapshotTestsErrorLongTextLatin];
```

Comment:
This snapshot is cropped a bit which may mean we're missing some information from the test. I've filed https://github.com/material-components/material-components-ios/issues/5942 to track being able to customize the padding of the contained view.

Reply:
It's not cropped. It's capturing an existing bug.

Reply:
That text would effectively go off-screen for users.

Reply:
Ah I see. In this case if we're testing the manner by which the content is cropped it may benefit us to customize the container view to show what the expected buggy behavior looks like. E.g. have left/right margins that match the expected distance to a container view (or the screen).

Reply:
I think the expectation is that it doesn't go off-screen. So once the bug is fixed the padding would be correct.

Reply:
I also think the focus of this PR should be to get ANY tests in for Outlined controllers and prepare the way for Text Fields in general. I don't want to delay this PR by trying to figure out why custom padding wasn't resulting in identical screenshots.  I had already attempted #5942 and there were screen diffs that weren't visible to my eyes.